content/en/blog/_posts/2025-12-17-kubernetes-v1-35-release/index.md

```diff
@@ -0,0 +1,450 @@
+
+The kubelet can now inject standard topology labels, such as `topology.kubernetes.io/zone` and `topology.kubernetes.io/region`, into Pods as environment variables or projected volume files. The primary benefit is a safer and more efficient way for workloads to be topology-aware. It allows applications to natively adapt to their availability zone or region without dependencies on the API server, strengthening security by upholding the principle of least privilege and simplifying cluster configuration. 
+
+This work was done as part of [KEP #4742](https://kep.k8s.io/4742) led by SIG Node.
+
+### Move storage version migrator in-tree
+
+Kubernetes is improving how it handles storage migrations by introducing a built-in, streamlined mechanism that makes it easy for users to rewrite data that needs updating for maintenance activities such as schema evolution and changes in encryption at rest. Previously, the common approach was to trigger no-op updates, often by piping `kubectl get` into `kubectl replace`, but this was inefficient for large or numerous resources and especially burdensome for data-heavy objects like secrets. With the new storage migration capability, Kubernetes safely handles update conflicts and inconsistent continue tokens behind the scenes, since only the rewrite itself matters. This enhancement eliminates the need for manual scripting or deep operational knowledge, providing administrators with a simple, reliable, and automated way to ensure stored data remains current and properly maintained. 
+
+This work was done as part of [KEP #4912](https://kep.k8s.io/4912) led by SIG API Machinery.
+
+### Mutable CSINode allocatable property
+
+A CSI (Container Storage Interface) driver is a Kubernetes plugin that provides a consistent way for storage systems to be exposed to containerized workloads. The `CSINode` object records details about all CSI drivers installed on a node. However, a mismatch can arise between the reported and actual attachment capacity on nodes. When volume slots are consumed after a CSI driver starts up, `kube-scheduler` may assign stateful pods to nodes without sufficient capacity, ultimately getting stuck in a `ContainerCreating` state.
+
+This KEP makes `CSINode.spec.drivers[*].allocatable.count` mutable so that a node’s available volume attachment capacity can be updated dynamically. It also allows CSI drivers to control how frequently the `allocatable.count` value is updated on all nodes by introducing a configurable refresh interval, defined through the `CSIDriver` object. Additionally, it automatically updates `CSINode.spec.drivers[*].allocatable.count` on detecting a failure in volume attachment due to insufficient capacity. Although this feature graduated to Beta in v1.34, it continues to be in Beta for v1.35 to allow time for feedback. 
```

Comment:
```suggestion
This KEP makes `CSINode.spec.drivers[*].allocatable.count` mutable so that a node’s available volume attachment capacity can be updated dynamically. It also allows CSI drivers to control how frequently the `allocatable.count` value is updated on all nodes by introducing a configurable refresh interval, defined through the `CSIDriver` object. Additionally, it automatically updates `CSINode.spec.drivers[*].allocatable.count` on detecting a failure in volume attachment due to insufficient capacity. Although this feature graduated to Beta in v1.34, it remains in Beta for v1.35 to allow time for feedback. 
```